packages/office-ui-fabric-react/src/utilities/test/TestUtilities.ts

```diff
@@ -0,0 +1,33 @@
+import { ReactWrapper } from 'enzyme';
+import * as sinon from 'sinon';
+/**
+ * Injects a function call prior to running componentDidUpdate for a component
+ * rendered using enzyme deep rendering.
+ * @param wrapper The enzyme deep rendering wrapper whose onComponentDidUpdate
+ * @param fn The function to run prior to onComponentDidUpdate
+ */
+export function runPriorToComponentDidUpdate(wrapper: ReactWrapper<any, any>, fn: () => void) {
+  // tslint:disable-next-line:no-string-literal
+  const originalComponentDidUpdate = wrapper.instance()['componentDidUpdate'];
+
+  if (typeof originalComponentDidUpdate !== 'function') {
+    throw new Error('Tried to override componentDidUpdate on a ReactWrapper that does not have that function');
+  }
+
+  // tslint:disable-next-line:no-string-literal
+  wrapper.instance()['componentDidUpdate'] = function (prevProps: any) {
+    fn();
+    originalComponentDidUpdate.call(this, prevProps);
```

Comment:
componentDidUpdate takes in both `prevProps` and `prevState`. But you can probably just make this generic;

obj[methodName] = (...args) {
  fn();
  obj[methodName].apply(this, args);
}